core/ir/x86/encode.c

```diff
@@ -3021,7 +3021,8 @@ encode_cti(instr_t *instr, byte *copy_pc, byte *final_pc,
     } else {
         /* 32-bit offset */
         /* offset is from start of next instr */
-        ptr_int_t offset = target - ((ptr_int_t)(pc + 4 - copy_pc + final_pc));
+        ptr_int_t offset =
+            target - ((ptr_int_t)(pc + 4 - copy_pc + (ptr_uint_t)final_pc));
```

Comment:
Should line 3015 also have `(ptr_uint_t)final_pc`?

Reply:
Added, but I didn't detect this one